### PR TITLE
A ton of stuff

### PR DESCRIPTION
**Pull request by tudorluca**
Review the code and let's talk over it. :)

### CODE REVIEW

**Comment by tudorluca** [comment]
I also use anko's toast(message) function :).

File: chapter4/src/main/java/com/backstopmedia/kotlin/chapter4/api/KTwitterApiClient.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+/**
+ * Created by Tudor Luca on 14/10/15.
+ */
+class KTwitterApiClient(val session: TwitterSession) : TwitterApiClient(session) {
+
+    val kTwitterApi: KTwitterApi by lazy { getService(KTwitterApi::class.java) }
+}
+
+interface KTwitterApi {
```
Comment (asarazan) [inline chapter4/src/main/java/com/backstopmedia/kotlin/chapter4/api/KTwitterApiClient.kt:17]: Is there an existing twitter API class we can piggyback off with extensions? This feels messy.

File: chapter4/build.gradle [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    }
+}
+
+dependencies {
+    compile fileTree(dir: 'libs', include: ['*.jar'])
+    compile 'com.android.support:appcompat-v7:23.0.1'
+    compile 'com.android.support:design:23.0.1'
+    compile 'com.android.support:recyclerview-v7:23.0.1'
+    compile 'com.android.support:cardview-v7:23.0.1'
+    compile 'com.github.bumptech.glide:glide:3.6.1'
```
Comment (asarazan) [inline chapter4/build.gradle:38]: I think everybody cast their lot for Picasso as the image loading lib

**Comment by asarazan** [comment]
Also we need to upgrade these icons to at least xxhpdi-- they look blurry as hell :-P

File: chapter4/src/main/java/com/backstopmedia/kotlin/chapter4/api/KTwitterApiClient.kt (hunk 1 above)
Comment (tudorluca) [inline chapter4/src/main/java/com/backstopmedia/kotlin/chapter4/api/KTwitterApiClient.kt:17]: Nope. This is how twitter enforces you to do things and extend the Retrofit api.  https://docs.fabric.io/android/twitter/access-rest-api.html

File: chapter4/build.gradle (hunk 2 above)
Comment (tudorluca) [inline chapter4/build.gradle:38]: Hahaha. Join the dark side... Join it! :)

File: chapter4/src/main/java/com/backstopmedia/kotlin/chapter4/api/KTwitterApiClient.kt (hunk 1 above)
Comment (asarazan) [inline chapter4/src/main/java/com/backstopmedia/kotlin/chapter4/api/KTwitterApiClient.kt:17]: Looks like the home timeline is built into the SDK, but I guess if we want an observable this is probably the easiest way to do it.

Really wish Retrofit 2.0 was out so we could just add a bunch of extensions to the `Call` object :-(

**Comment by tudorluca** [comment]
I copied them from an old project. Maybe you can find better ones. They look too similar :)

File: chapter4/src/main/java/com/backstopmedia/kotlin/chapter4/ui/NavigationHelper.kt [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+            navigationHelper.setContentViewForActivity(activity, resId)
+            return navigationHelper
+        }
+    }
+
+    private val interactor: UserInteractor by lazy { UserInteractorImpl(TwitterCore.getInstance().sessionManager.activeSession) }
+    private val presenter: NavigationDrawerPresenter  by lazy { NavigationDrawerPresenterImpl(interactor) }
+
+    private var isSubdecorInstalled = false
+    private var drawerLayout: DrawerLayout by Delegates.notNull()
```
Comment (nbilyk) [inline chapter4/src/main/java/com/backstopmedia/kotlin/chapter4/ui/NavigationHelper.kt:50]: In this case do we want Delegates.notNull() or would it be better to use the lateinit keyword?

Comment (pawegio) [inline chapter4/src/main/java/com/backstopmedia/kotlin/chapter4/ui/NavigationHelper.kt:50]: discussed on slack: https://kotlinbook.slack.com/archives/general/p1444930972000060

**Comment by asarazan** [comment]
Alright lets merge this shit.

**Pull request merged**